core/src/main/scala/org/apache/spark/rdd/NewHadoopRDD.scala

```diff
@@ -244,7 +244,13 @@ class NewHadoopRDD[K, V](
         try {
           val _reader = format.createRecordReader(
             split.serializableHadoopSplit.value, hadoopAttemptContext)
-          _reader.initialize(split.serializableHadoopSplit.value, hadoopAttemptContext)
+          try {
```

Comment:
can we use `Utils.tryWithResource` instead?

Reply:
I don't think that has the right semantics: we only want to close on error, whereas I think `tryWithResource` will close at the end of the try block no matter what.

Maybe I can add another util to clean this up?